lib/std/time.zig

```diff
@@ -59,6 +59,398 @@ pub fn sleep(nanoseconds: u64) void {
     std.os.nanosleep(s, ns);
 }
 
+pub const TimeZone = @import("time/TimeZone.zig");
+
+pub const LocalTimeError = error{
+    /// The local time zone of the system could not be determined.
+    LocalTimeUnavailable,
+} || std.mem.Allocator.Error;
+
+pub fn localtime(allocator: std.mem.Allocator) LocalTimeError!TimeZone {
+    switch (builtin.os.tag) {
+        .linux => {
+            var tz_file = std.fs.openFileAbsolute("/etc/localtime", .{}) catch
+                return error.LocalTimeUnavailable;
+            defer tz_file.close();
+
+            var buffered = std.io.bufferedReader(tz_file.reader());
+            return TimeZone.parse(allocator, buffered.reader()) catch
+                return error.LocalTimeUnavailable;
+        },
+        else => @compileError("`localtime` has not been implemented for this OS!"),
+    }
+}
+
+/// ISO 8601 compliant date representation.
+pub const Date = struct {
+    /// milliseconds, range 0 to 999
+    millisecond: u16,
+
+    /// seconds, range 0 to 60
+    second: u8,
+
+    /// minutes, range 0 to 59
+    minute: u8,
+
+    /// hours, range 0 to 23
+    hour: u8,
+
+    /// day of the month, range 1 to 31
+    day: u8,
+
+    /// day of the year, range 1 to 366
+    year_day: u16,
+
+    /// day of the week, enum Monday to Sunday
+    week_day: Weekday,
+
+    /// month, enum January to December
+    month: Month,
+
+    /// year, year 0000 is equal to 1 BCE
+    year: i32,
+
+    utc_timestamp: i64,
+    tt: TimeZone.TimeType,
+
+    pub const Weekday = enum(u3) {
+        monday = 1,
+        tuesday = 2,
+        wednesday = 3,
+        thursday = 4,
+        friday = 5,
+        saturday = 6,
+        sunday = 7,
+
+        pub const names = [_][]const u8{
+            "Monday",
+            "Tuesday",
+            "Wednesday",
+            "Thursday",
+            "Friday",
+            "Saturday",
+            "Sunday",
+        };
+
+        pub fn shortName(w: Weekday) []const u8 {
+            return names[@intFromEnum(w) - 1][0..3];
+        }
+
+        pub fn longName(w: Weekday) []const u8 {
+            return names[@intFromEnum(w) - 1];
+        }
+    };
+
+    pub const Month = enum(u4) {
+        january = 1,
+        february = 2,
+        march = 3,
+        april = 4,
+        may = 5,
+        june = 6,
+        july = 7,
+        august = 8,
+        september = 9,
+        october = 10,
+        november = 11,
+        december = 12,
+
+        pub const names = [_][]const u8{
+            "January",
+            "February",
+            "March",
+            "April",
+            "May",
+            "June",
+            "July",
+            "August",
+            "September",
+            "October",
+            "November",
+            "December",
+        };
```

Comment:
I feel like these need to be off in some Locale specific area; not as a `pub` field in the `Month` enum.

Reply:
Should these `names` constants be provided at all? They are specifically English and not localized and if you do not intend to localize your application to languages other than English it's not too difficult to do `@tagName` on the tag and then customize the capitalization as well. Maybe I want to capitalize it "january" or "JANUARY" instead.